Add --keyField parameter

olsonanl opened the pull request:
Hi Bruce - what do you think of this change? I used it to map a file of NCBI genome accessions to BV-BRC data. I don't know if there's an existing mechanism to do this, but changing the key field like this seems pretty clean.
Thanks,
Bob

olsonanl committed:
Add --keyField parameter

bparrello commented:
We have a similar need for features, and we use the p3-find-features script, so I added p3-find-genomes. This has been pushed to GitHub and will be absorbed in the currently-outstanding pull request.

The advantage is that I can restrict the key fields to those that are indexed, and turn the batching on and off on a key-by-key basis. (So, for example, if we added "genus" as a key, we'd put it in slow mode instead of batch mode.) We already have this logic for features.

olsonanl commented:
Very cool. I thought we had such a find but didn't twig that it was just for features. Thanks much. Now the tedious process of recreating all the CLI packages...
________________________________
From: Bruce Parrello ***@***.***>
Sent: Tuesday, April 1, 2025 9:11 AM
To: BV-BRC/p3_cli ***@***.***>
Cc: Olson, Robert D. ***@***.***>; Author ***@***.***>
Subject: Re: [BV-BRC/p3_cli] Add --keyField parameter (PR #13)

We have a similar need for features, and we use the p3-find-features script, so I added p3-find-genomes. This has been pushed to GitHub and will be absorbed in the currently-outstanding pull request. The advantage is that I can restrict the
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.

ZjQcmQRYFpfptBannerEnd

We have a similar need for features, and we use the p3-find-features script, so I added p3-find-genomes. This has been pushed to GitHub and will be absorbed in the currently-outstanding pull request.

The advantage is that I can restrict the key fields to those that are indexed, and turn the batching on and off on a key-by-key basis. (So, for example, if we added "genus" as a key, we'd put it in slow mode instead of batch mode.) We already have this logic for features.

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v3/__https://github.com/BV-BRC/p3_cli/pull/13*issuecomment-2769502910__;Iw!!G_uCfscf7eWS!fkyEJephq6gt5YvYcFnFoPsmRkRlHF0S-ovdGy7pVkc1I327MvHGXsCm__j2gpv-HKdVUoOqaTm5nmqW-Pj7W_2gdA$>, or unsubscribe<https://urldefense.us/v3/__https://github.com/notifications/unsubscribe-auth/AAQNOIXD3FODGDVEIYD7QRL2XKNB7AVCNFSM6AAAAAB2EZIRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRZGUYDEOJRGA__;!!G_uCfscf7eWS!fkyEJephq6gt5YvYcFnFoPsmRkRlHF0S-ovdGy7pVkc1I327MvHGXsCm__j2gpv-HKdVUoOqaTm5nmqW-PhWQnyNRQ$>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

[bparrello]bparrello left a comment (BV-BRC/p3_cli#13)<https://urldefense.us/v3/__https://github.com/BV-BRC/p3_cli/pull/13*issuecomment-2769502910__;Iw!!G_uCfscf7eWS!fkyEJephq6gt5YvYcFnFoPsmRkRlHF0S-ovdGy7pVkc1I327MvHGXsCm__j2gpv-HKdVUoOqaTm5nmqW-Pj7W_2gdA$>

We have a similar need for features, and we use the p3-find-features script, so I added p3-find-genomes. This has been pushed to GitHub and will be absorbed in the currently-outstanding pull request.

The advantage is that I can restrict the key fields to those that are indexed, and turn the batching on and off on a key-by-key basis. (So, for example, if we added "genus" as a key, we'd put it in slow mode instead of batch mode.) We already have this logic for features.

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v3/__https://github.com/BV-BRC/p3_cli/pull/13*issuecomment-2769502910__;Iw!!G_uCfscf7eWS!fkyEJephq6gt5YvYcFnFoPsmRkRlHF0S-ovdGy7pVkc1I327MvHGXsCm__j2gpv-HKdVUoOqaTm5nmqW-Pj7W_2gdA$>, or unsubscribe<https://urldefense.us/v3/__https://github.com/notifications/unsubscribe-auth/AAQNOIXD3FODGDVEIYD7QRL2XKNB7AVCNFSM6AAAAAB2EZIRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRZGUYDEOJRGA__;!!G_uCfscf7eWS!fkyEJephq6gt5YvYcFnFoPsmRkRlHF0S-ovdGy7pVkc1I327MvHGXsCm__j2gpv-HKdVUoOqaTm5nmqW-PhWQnyNRQ$>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

bparrello commented:
I implemented it using a new script, p3-find-genome, to parallel p3-find-feature. It’s part of the umpteen changes pending with pull request 12.

 

From: olsonanl ***@***.***> 
Sent: Thursday, April 24, 2025 1:20 PM
To: BV-BRC/p3_cli ***@***.***>
Cc: Bruce Parrello ***@***.***>; Assign ***@***.***>
Subject: Re: [BV-BRC/p3_cli] Add --keyField parameter (PR #13)

 

  <https://avatars.githubusercontent.com/u/2152226?s=20&v=4> olsonanl left a comment (BV-BRC/p3_cli#13) <https://github.com/BV-BRC/p3_cli/pull/13#issuecomment-2828505849> 

Is there a PR somewhere for this? I'm not finding it

—
Reply to this email directly, view it on GitHub <https://github.com/BV-BRC/p3_cli/pull/13#issuecomment-2828505849> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB652NH3GR2O3HBBGP2DJNT23ETMXAVCNFSM6AAAAAB2EZIRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMRYGUYDKOBUHE> .
You are receiving this because you were assigned.  <https://github.com/notifications/beacon/AB652NDNS3NEL5IG53U4QZ323ETMXA5CNFSM6AAAAAB2EZIRZ2WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVIS6JPS.gif> Message ID: ***@***.*** ***@***.***> >

olsonanl commented:
Got it - I saw that one but saw Comparative Systems and was thinking it was something else. Thanks.

That's all ready to merge?

bparrello commented:
Yes, absolutely.

 

From: olsonanl ***@***.***> 
Sent: Friday, April 25, 2025 8:38 AM
To: BV-BRC/p3_cli ***@***.***>
Cc: Bruce Parrello ***@***.***>; Assign ***@***.***>
Subject: Re: [BV-BRC/p3_cli] Add --keyField parameter (PR #13)

 

  <https://avatars.githubusercontent.com/u/2152226?s=20&v=4> olsonanl left a comment (BV-BRC/p3_cli#13) <https://github.com/BV-BRC/p3_cli/pull/13#issuecomment-2830462877> 

Got it - I saw that one but saw Comparative Systems and was thinking it was something else. Thanks.

That's all ready to merge?

—
Reply to this email directly, view it on GitHub <https://github.com/BV-BRC/p3_cli/pull/13#issuecomment-2830462877> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB652NHBBCA67QQ4FH64HNL23I3FFAVCNFSM6AAAAAB2EZIRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMZQGQ3DEOBXG4> .
You are receiving this because you were assigned.  <https://github.com/notifications/beacon/AB652NDKFPH4YVIUPR6HTJ323I3FFA5CNFSM6AAAAAB2EZIRZ2WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVIWVXZ2.gif> Message ID: ***@***.*** ***@***.***> >

olsonanl commented:
You can also show a compare regions for a list of features with something like this

https://www.bv-brc.org/view/FeatureList/?in(feature_id,(PATRIC.1671597.3.AP019695.CDS.2470054.2471238.fwd,PATRIC.2718527.9.CAVJCN010000145.CDS.1207.2349.rev,PATRIC.2486576.6.CAUWBR010000016.CDS.12232.13419.rev,PATRIC.592010.4.NZ_GG665859.CDS.184474.185649.fwd))#view_tab=compareRegionViewer

replacing feature_id with patric_id and using fig ids instead of the internal identifiers
________________________________
From: Bruce Parrello ***@***.***>
Sent: Monday, April 28, 2025 8:47 AM
To: BV-BRC/p3_cli ***@***.***>
Cc: Olson, Robert D. ***@***.***>; Author ***@***.***>
Subject: Re: [BV-BRC/p3_cli] Add --keyField parameter (PR #13)

bparrello left a comment (BV-BRC/p3_cli#13) Yes, absolutely. <br> <br> <br> <br> From: olsonanl ***@ ***. ***&gt; <br> Sent: Friday, April 25, 2025 8: 38 AM <br> To: BV-BRC/p3_cli ***@ ***. ***&gt; <br>
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.

ZjQcmQRYFpfptBannerEnd
[https://avatars.githubusercontent.com/u/8248628?s=20&v=4]bparrello left a comment (BV-BRC/p3_cli#13)<https://urldefense.us/v3/__https://github.com/BV-BRC/p3_cli/pull/13*issuecomment-2835307800__;Iw!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nmWpIMqrA$>
Yes, absolutely. <br> <br> <br> <br> From: olsonanl ***@***.***&gt; <br> Sent: Friday, April 25, 2025 8:38 AM <br> To: BV-BRC/p3_cli ***@***.***&gt; <br> Cc: Bruce Parrello ***@***.***&gt;; Assign ***@***.***&gt; <br> Subject: Re: [BV-BRC/p3_cli] Add --keyField parameter (PR #13) <br> <br> <br> <br> &lt;https://avatars.githubusercontent.com/u/2152226?s=20&amp;v=4&gt;<https://urldefense.us/v3/__https://avatars.githubusercontent.com/u/2152226?s=20&amp;v=4&gt;__;!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nmG0etxig$> olsonanl left a comment (BV-BRC/p3_cli#13) &lt;https://github.com/BV-BRC/p3_cli/pull/13#issuecomment-2830462877&gt;<https://urldefense.us/v3/__https://github.com/BV-BRC/p3_cli/pull/13*issuecomment-2830462877&gt;__;Iw!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nm0bxzPqw$> <br> <br> Got it - I saw that one but saw Comparative Systems and was thinking it was something else. Thanks. <br> <br> That&#39;s all ready to merge? <br> <br> — <br> Reply to this email directly, view it on GitHub &lt;https://github.com/BV-BRC/p3_cli/pull/13#issuecomment-2830462877&gt;<https://urldefense.us/v3/__https://github.com/BV-BRC/p3_cli/pull/13*issuecomment-2830462877&gt;__;Iw!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nm0bxzPqw$> , or unsubscribe &lt;https://github.com/notifications/unsubscribe-auth/AB652NHBBCA67QQ4FH64HNL23I3FFAVCNFSM6AAAAAB2EZIRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMZQGQ3DEOBXG4&gt;<https://urldefense.us/v3/__https://github.com/notifications/unsubscribe-auth/AB652NHBBCA67QQ4FH64HNL23I3FFAVCNFSM6AAAAAB2EZIRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMZQGQ3DEOBXG4&gt;__;!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nk0eVkJaw$> . <br> You are receiving this because you were assigned. &lt;https://github.com/notifications/beacon/AB652NDKFPH4YVIUPR6HTJ323I3FFA5CNFSM6AAAAAB2EZIRZ2WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVIWVXZ2.gif&gt;<https://urldefense.us/v3/__https://github.com/notifications/beacon/AB652NDKFPH4YVIUPR6HTJ323I3FFA5CNFSM6AAAAAB2EZIRZ2WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVIWVXZ2.gif&gt;__;!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nm_isrrCw$> Message ID: ***@***.*** ***@***.***&gt; &gt; <br> <br>

—
Reply to this email directly, view it on GitHub<https://urldefense.us/v3/__https://github.com/BV-BRC/p3_cli/pull/13*issuecomment-2835307800__;Iw!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nmWpIMqrA$>, or unsubscribe<https://urldefense.us/v3/__https://github.com/notifications/unsubscribe-auth/AAQNOITSADNZLDIZPIAIAZD23YWPVAVCNFSM6AAAAAB2EZIRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMZVGMYDOOBQGA__;!!G_uCfscf7eWS!fAlxy_JmnQFl_mweBev3l3pFQs7DqLMd5UPnCnRg59DYXN0L3YethzL8Y4fTX91fmpGmAdCEQwL2WK8i-nn0j1Nw4A$>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
